api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1857,8 +1857,98 @@ type AzurePlatformSpec struct {
+// and data plane components that authenticate with Azure's API.
+type AzureResourceManagedIdentities struct {
+	// ControlPlane contains the client IDs of all the managed identities on the HCP control plane needing to
+	// authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlane ControlPlaneManagedIdentities `json:"controlPlane"`
+
+	// Future placeholder - DataPlaneMIs * DataPlaneManagedIdentities
+}
+
+// ManagedIdentityClientID is a client ID of a managed identity
+// +kubebuilder:validation:XValidation:rule="self.matches('^[{]?[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}[}]?$')",message="the client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters in the form 8-4-4-4-12."
+type ManagedIdentityClientID string
+
+// ControlPlaneManagedIdentities contains the client IDs of all the managed identities on the HCP control plane needing
+// to authenticate with Azure's API.
+// Managed identity regex pattern is from Microsoft here - https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftmanagedidentity.
+// The format a managed identity should be `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{managedIdentityName}`.
+type ControlPlaneManagedIdentities struct {
+	// cloudProviderClientID is the client ID of a pre-existing managed identity ID associated with the azure
+	// cloud provider, aka ccm. The client ID of a managed identity must be a valid UUID. It should be 5 groups of
+	// hyphen separated hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	CloudProviderClientID ManagedIdentityClientID `json:"cloudProviderClientID"`
+
+	// clusterAPIAzureClientID is the client ID of a pre-existing managed identity ID associated with cluster-api
+	// azure. The client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated
+	// hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	ClusterAPIAzureClientID ManagedIdentityClientID `json:"clusterAPIAzureClientID"`
+
+	// controlPlaneClientID is the client ID of a pre-existing managed identity ID associated with the control plane
+	// operator. The client ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated
+	// hexadecimal characters in the form 8-4-4-4-12.
+	//
+	// +kubebuilder:validation:Required
+	ControlPlaneClientID ManagedIdentityClientID `json:"controlPlaneClientID"`
+
+	// kmsClientID is the client ID of a pre-existing managed identity ID associated with Azure KMS. The client
+	// ID of a managed identity must be a valid UUID. It should be 5 groups of hyphen separated hexadecimal characters
+	// in the form 8-4-4-4-12.
+	//
+	// +optional
```

Comment:
Why is this one marked as optional?

Reply:
KMS is normally optional on HCP and not required

Reply:
Doesn't the KMS identity become *required* if KMS is leveraged?

Looking at the Hypershift API of HostedCluster, for the case of AWS STS I don't see the KMS identity being defined in `.platform.aws.rolesRef` AWSPlatformSpec. Instead, it seems to be defined at the the functionality usage point. This is, in `.secretEncryption.kms.aws.auth.awsKms`.

This seems to be inconsistent. Are we aware of that difference? Is there some reason for it?
Is hypershift validating that it is "optional", but "required" when kms is/will be used for the cluster being created?

Reply:
Yeah that does seem inconsistent. I was not aware of the difference. 

I'm inclined to remove the managed identity that is needed here out of the HC and into `AzureKMSSpec`. WDYT @muraee since you worked on the KMS implementation?

Reply:
You could use a CEL validation if they are on the same object to make sure that they are either both supplied, or neither are supplied

Reply:
yeah I think it makes more sense to move this field to `AzureKMSSpec` and stay consistent.